.github/workflows/actions-ci.yml

```diff
@@ -259,13 +259,49 @@ jobs:
       - name: Run tests
         run: cmake --build ./build --target run_tests
 
+  gcc-13-pedantic:
+    if: github.repository_owner == 'aws'
+    needs: [ sanity-test-run ]
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup CMake
+        uses: threeal/cmake-action@v1.3.0
+        with:
+          generator: Ninja
+          c-compiler: gcc-13
+          cxx-compiler: g++-13
+          options: CMAKE_BUILD_TYPE=Release CMAKE_C_FLAGS=-pedantic CMAKE_CXX_FLAGS=-pedantic
+      - name: Build Crypto
+        run: cmake --build ./build --target crypto
+      - name: Build SSL
+        run: cmake --build ./build --target ssl
+
+  clang-18-pedantic:
```

Comment:
do we need distinct CI jobs for pedantry? can we add these flags to existing GCC 13 / clang 18 jobs?

Reply:
Supporting a "pedantic" build can very painful/restrictive. With this PR we are able to compile the "crypto" and "ssl" build targets with `-pedantic`, but we can't build/run the tests against it. 

Since our tests are a better indicator for the health of a build than pedantic warnings, I chose to limit the use of `-pedantic` here for just helping ensure that [downstream consumers](https://github.com/aws/aws-lc-rs/issues/425)) aren't blocked b/c they have a build environment that requires `-pedantic`.

I used only the latest versions of GCC and Clang since those would likely produce a superset of the pedantic warnings from their previous versions.